app/services/set_decline_by_default.rb

```diff
@@ -0,0 +1,38 @@
+class SetDeclineByDefault
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call
+    return false if pending_application_choices.any? || application_choices.none?(&:offered_at)
+
+    most_recent_decision_date = [
+      application_choices.maximum(:offered_at),
+      application_choices.maximum(:rejected_at),
+    ].compact.max
+
+    dbd_days = TimeLimitCalculator.new(
+      rule: :decline_by_default,
+      effective_date: most_recent_decision_date,
+    ).call
+
+    dbd_date = dbd_days.business_days.after(most_recent_decision_date).end_of_day
+
+    application_form.application_choices.where.not(offered_at: nil).update_all(
+      decline_by_default_at: dbd_date,
+      decline_by_default_days: dbd_days,
+    )
+  end
+
+private
+
+  attr_reader :application_form
+
+  def application_choices
+    application_form.application_choices
+  end
+
+  def pending_application_choices
```

Comment:
We're missing withdrawls in this implementation, because we haven't build that yet. Might be worth documenting somewhere that once we implement that we need to trigger `SetDeclineByDefault` in the service, and update the `pending_application_choices` to include things in withdrawn state.

Reply:
We hadn't considered withdrawals up to now, but funnily enough the refactor I discussed with Steve earlier will solve this neatly. If check 1 (see above) around pending choices is checking that ```status == 'awaiting_provider_decision'```, then any withdrawn applications will be regarded automatically as decided upon/non-pending. If check 2 is based on ```status == 'offer'```, then DBD will not be set on withdrawn applications (if DBD fields have already been set before withdrawal, the sweeper task just needs to make sure it declines only application_choices in 'offer' state and it will be fine).

So, all we need is a note somewhere to call SetDeclineByDefault service from the withdrawal service.